xds/csds/csds_e2e_test.go

```diff
@@ -361,50 +390,201 @@ func (s) TestCSDS(t *testing.T) {
 	if err := checkClientStatusResponse(ctx, stream, wantResp); err != nil {
 		t.Fatal(err)
 	}
+}
 
-	// Update the first resource of each type in the management server to a
-	// value which is expected to be NACK'ed by the xDS client.
-	listeners[0].ApiListener = &v3listenerpb.ApiListener{}
-	routes[0].VirtualHosts = []*v3routepb.VirtualHost{{Routes: []*v3routepb.Route{{}}}}
-	clusters[0].ClusterDiscoveryType = &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_STATIC}
-	endpoints[0].Endpoints = []*v3endpointpb.LocalityLbEndpoints{{}}
+// Tests CSDS functionality. The test performs the following:
+//   - Spins up a management server and creates two xDS clients talking to it.
+//   - Registers one watch on each xDS client, and verifies that the CSDS
+//     response reports resources in REQUESTED state.
```

Comment:
In between REQUESTED and NACK you also test if both of them are in ACKED state.

Reply:
Done.